core/objects.go

```diff
@@ -249,6 +250,14 @@ func (ch Challenge) RecordsSane() bool {
 			return false
 		}
 		return true
+	case ChallengeTypeDNSAccount01:
```

Comment:
Lines 254 - 260 are identical to the dns-01 case above. You could remove these lines and rewrite 245 as 
```
	case ChallengeTypeDNS01, ChallengeTypeDNSAccount01:
```